src/Integration/LocalServices/SolutionBindingInformationProvider.cs

```diff
@@ -31,13 +31,46 @@ public SolutionBindingInformationProvider(IServiceProvider serviceProvider)
         }
 
         #region ISolutionBindingInformationProvider
+        public bool IsSolutionBound()
+        {
+            return this.GetSolutionBinding() != null;
+        }
+
         public IEnumerable<Project> GetBoundProjects()
         {
+            return this.GetBoundProjects(this.GetSolutionBinding());
+        }
+
+        public IEnumerable<Project> GetUnboundProjects()
+        {
+            return this.GetUnBoundProject(this.GetSolutionBinding());
```

Comment:
Because I still need the private methods since I call them and pass in the instance of solution binding information which I read once either at the beginning of GetBoundProjects or at the beginning of GetUnboundProjects (public). The private methods get the binding info and (re)use it throughout the logic.